File: Python/1672.py
# 1672. Richest Customer Wealth
# https://leetcode.com/problems/richest-customer-wealth/

class Solution:
  def maximumWealth(self, accounts: List[List[int]]) -> int:
    return max(list(map(lambda acc: (reduce(lambda a, b: a + b, acc, 0)), accounts)))

File: Rust/1672.rs
/**
 * 1672. Richest Customer Wealth
 * https://leetcode.com/problems/richest-customer-wealth/
 */

impl Solution {
  pub fn maximum_wealth(accounts: Vec<Vec<i32>>) -> i32 {
    pub fn sum(account: Vec<i32>) -> i32 {
      return account.iter().fold(0, |a, b| a + b);
    }

    let account_sum: Vec<i32> = accounts
      .iter()
      .map(|x| sum(x.to_vec()))
      .collect::<Vec<_>>();

    return *account_sum.iter().max().unwrap();
  }
}
